File: app/src/main/java/sk/styk/martin/apkanalyzer/activity/dialog/SimpleTextDialog.java
package sk.styk.martin.apkanalyzer.activity.dialog;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import sk.styk.martin.apkanalyzer.R;

/**
 * @author Martin Styk
 * @version 23.06.2017.
 */
public class SimpleTextDialog extends DialogFragment {

    private static final String ARG_TITLE = "title";
    private static final String ARG_MESSAGE = "value";

    public static SimpleTextDialog newInstance(String title, String message) {
        SimpleTextDialog frag = new SimpleTextDialog();
        Bundle args = new Bundle();
        args.putString(ARG_TITLE, title);
        args.putString(ARG_MESSAGE, message);
        frag.setArguments(args);
        return frag;
    }


    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        String title = getArguments().getString(ARG_TITLE);
        String message = getArguments().getString(ARG_MESSAGE);

        return new AlertDialog.Builder(getActivity())
                .setTitle(title)
                .setMessage(message)
                .setNegativeButton(R.string.dismiss, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        dismiss();
                    }
                })
                .create();
    }


}

File: app/src/main/java/sk/styk/martin/apkanalyzer/activity/dialog/SimpleTextDialog.kt
package sk.styk.martin.apkanalyzer.activity.dialog

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog

import sk.styk.martin.apkanalyzer.R

/**
 * @author Martin Styk
 * @version 23.06.2017.
 */
class SimpleTextDialog : DialogFragment() {


    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val title = arguments.getString(ARG_TITLE)
        val message = arguments.getString(ARG_MESSAGE)

        return AlertDialog.Builder(activity)
                .setTitle(title)
                .setMessage(message)
                .setNegativeButton(R.string.dismiss) { dialogInterface, i -> dismiss() }
                .create()
    }

    companion object {

        private val ARG_TITLE = "title"
        private val ARG_MESSAGE = "value"

        fun newInstance(title: String, message: String): SimpleTextDialog {
            val frag = SimpleTextDialog()
            val args = Bundle()
            args.putString(ARG_TITLE, title)
            args.putString(ARG_MESSAGE, message)
            frag.arguments = args
            return frag
        }
    }


}
